staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -169,6 +169,8 @@ func (s *store) Create(ctx context.Context, key string, obj, out runtime.Object,
 	).Then(
 		clientv3.OpPut(key, string(newData), opts...),
 	).Commit()
+	// We count all operations that happened in the same lease, regardless of success or failure.
+	s.leaseManager.UpdateLeaseObjectCount(1)
```

Comment:
How about increasing the counter inside of GetLease? This will be simpler.

Reply:
GetLease may success but the function may returned before transaction, see L163 
https://github.com/kubernetes/kubernetes/blob/bb376f161640529f600203faee4cd0de8ec16778/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L156-L171

Reply:
Increase the counter inside GetLease requires the caller to attach only one object, also implicit requires caller should not abort after GetLease before creating transactions. For example:
https://github.com/kubernetes/kubernetes/blob/bb376f161640529f600203faee4cd0de8ec16778/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L156-L164


L163 appears after GetLease but it may abort the function. Then the counter +1 inside GetLease, but nothing attached to the lease.

Reply:
I see your point.

I guess the transformer.TransformToStorage error shouldn't be that common, so I guess for the sake of simplicity, it should be fine to ignore this case and simply count inside.

Please note that if GetLease would increase counter, but the caller will not use that value, the risk is that we will create slightly more leases than needed.

On the other hand, my concern is that if we someone changes this code and forgets to all UpdateLeaseObjectCount, we may have completely broken functionality here.

Reply:
Change the order of transformer.TransformToStorage and ttlOpts (GetLease) are simple, my concern is further changes may forget this implicit behavior too. But you are right, count more lease objects or creates more leases is acceptable too.